### ISSUE TITLE
Usage of callbacks__

### ISSUE THREAD

**Issue by sim-san** [adriangb/scikeras#232]
Can you explain the usage of the routed parameter `callbacks__` from [Routed parameters](https://www.adriangb.com/scikeras/stable/advanced.html#routed-parameters) ?

- Is this a shortcut for the `fit__callbacks` ?
- Or can it be used to set parameters for each instantiated callback like the [example:  losses with routed parameters](https://www.adriangb.com/scikeras/stable/advanced.html#example-multiple-losses-with-routed-parameters) ?

I couldn't find a usage of the `route_params()` function with `destination="callbacks"`. Is this functionality already implemented ?

@stsievert

**Comment by adriangb** [comment]
Hi, thanks for the question.

> Can you explain the usage of the routed parameter callbacks__ from Routed parameters ?

This is the intended usage:

```python
import tensorflow as tf
from sklearn.datasets import make_classification
from scikeras.wrappers import KerasClassifier

X, y = make_classification(1000, 20, n_informative=10, random_state=0)

def get_clf(meta) -> tf.keras.Model:
    n_features_in_ = meta["n_features_in_"]
    n_classes_ = meta["n_classes_"]
    model = tf.keras.models.Sequential()
    model.add(tf.keras.layers.InputLayer((n_features_in_,)))
    model.add(tf.keras.layers.Dense(n_classes_, activation="softmax"))
    return model

clf = KerasClassifier(
    get_clf,
    loss="sparse_categorical_crossentropy",
    epochs=50,
    metrics=["acc"],
    callbacks=[tf.keras.callbacks.EarlyStopping],
    callbacks__0__monitor="val__acc",
    callbacks__0__min_delta=0.01,
    callbacks__0__patience=5,
    fit__validation_split=0.1,
)
clf.fit(X, y)  # stops after a couple iterations
```

Much more relevant for something like a learning rate scheduler, which you might actually want to have access to via hyperparameter tuning. If you don't need to tune any parameters to your callback, you can just use the Keras syntax of passing an object:

```python
clf = KerasClassifier(
    get_clf,
    loss="sparse_categorical_crossentropy",
    epochs=50,
    metrics=["acc"],
    callbacks=[tf.keras.callbacks.EarlyStopping(monitor="val_acc", ...)],
    fit__validation_split=0.1,
)
```

> Is this a shortcut for the fit__callbacks

Yep. `fit__callbacks` is the same as `callbacks`. These "shortcuts" are defined [here](https://github.com/adriangb/scikeras/blob/1fa934109382ffd0b9d0cde4f74fd15c83b2d3c6/scikeras/wrappers.py#L142-L157).

> Or can it be used to set parameters for each instantiated callback like the example: losses with routed parameters ?

As per above, yes the goal is that you can set parameters for each callback independently via the constructor so that each individual parameter can be tuned via a hyperparameter tuning routine.

> Is this functionality already implemented ?

Sadly it is not, it kind of dropped off the radar since loss/optimizer tuning were more important and less complex. Let me know if the above usage fits in with what you'd envision and I can implement it.

**Comment by adriangb** [comment]
I opened #233 as a draft to support the usage described above.

**Comment by sim-san** [comment]
> Sadly it is not, it kind of dropped off the radar since loss/optimizer tuning were more important and less complex. Let me know if the above usage fits in with what you'd envision and I can implement it.

I like the usage you describe above and I am looking forward to using it.

**Comment by sim-san** [comment]
Is there a schedule for when the next release will be released with this feature (https://github.com/adriangb/scikeras/pull/233) ?

**Comment by adriangb** [comment]
No schedule, but I'll try to cut a release soon (next couple days). I've been punting on fixing a bunch of stuff that Tensorflow broke recently, I don't want to cut a release that won't work with v2.7.X.

**Comment by adriangb** [comment]
@sim-san v0.4.0 is published! https://github.com/adriangb/scikeras/releases/tag/v0.4.0
